lib/bundler/runtime.rb

```diff
@@ -37,7 +37,17 @@ def setup(*groups)
 
         Bundler.rubygems.mark_loaded(spec)
         load_paths = spec.load_paths.reject {|path| $LOAD_PATH.include?(path) }
-        $LOAD_PATH.unshift(*load_paths)
+
+        # See Gem::Specification#add_self_to_load_path (since RubyGems 1.8)
+        insert_index = Gem.load_path_insert_index
```

Comment:
The `unless` [here](https://github.com/bundler/bundler/commit/69acfd88#diff-36d67678b29b89b0dcfe3f3f09a5ed9dR32) was refactored to the `reject` later on.  This commit appears to be the origin commit for the intention.